.github/copilot-instructions.md

```diff
@@ -0,0 +1,517 @@
+- Always run at least basic tests: `python -m pytest tests/test_component.py`
+- Test sample project functionality: Start the sample project and make a request to verify components render correctly
+- Check that imports work: `python -c "import django_components; print('OK')"`
+
+## Common Tasks
+
+### Repository Structure
+- `src/django_components/` -- main package source code
+- `tests/` -- comprehensive test suite with 1000+ tests
+- `sampleproject/` -- working Django project demonstrating component usage
+- `docs/` -- documentation source (uses mkdocs)
+- `requirements-dev.txt` -- development dependencies (validated to work)
+- `requirements-docs.txt` -- documentation building dependencies
+- `pyproject.toml` -- package configuration and dependencies
+- `tox.ini` -- test environment configuration for multiple Python/Django versions
+
+### Key Files to Check When Making Changes
+- Always check the sample project works after making changes to core functionality
+- Test component discovery by running `python manage.py components list` in the sample project
+- Verify component rendering by starting the sample project server and making requests
+- Check that import paths in `src/django_components/__init__.py` work correctly
+
+### CI/CD Information  
+- GitHub Actions workflow: `.github/workflows/tests.yml`
+- Tests run on Python 3.8-3.13 with Django 4.2-5.2
+- Includes Playwright browser testing (requires `playwright install chromium --with-deps`)
+- Documentation building uses mkdocs
+- Pre-commit hooks run black, isort, and flake8
+
+### Time Expectations
+- Installing dependencies: 1-2 minutes
+- Running basic component tests: 5 seconds
+- Running template tag tests (349 tests): 10 seconds  
+- Running full test suite: 2-5 minutes. NEVER CANCEL: Set timeout to 10+ minutes
+- Playwright browser install: 10+ minutes. NEVER CANCEL: Set timeout to 15+ minutes
+- Tox full test matrix: 10-30 minutes. NEVER CANCEL: Set timeout to 45+ minutes
+- Package building: 2-5 minutes, may timeout on network issues
+
+### Network Dependencies
+- pip installations may timeout due to network issues (this is environment-specific)
+- Playwright browser downloads may fail due to large file sizes
+- All core functionality works without additional network access once dependencies are installed
+
+### Development Workflow
+1. Install dependencies: `pip install -r requirements-dev.txt && pip install -e .`
+2. Make changes to source code in `src/django_components/`
+3. Run tests: `python -m pytest tests/test_component.py` (or specific test files)
+4. Run linting: `black src/django_components && isort src/django_components && flake8 .`
+5. Test sample project: `cd sampleproject && python manage.py runserver`
+6. Validate with curl: `curl http://127.0.0.1:8000/`
+7. Run broader tests before final commit: `python -m pytest tests/test_templatetags*.py`
+
+
+---
+
+<cursorrules>
```

Comment:
@EmilStenstrom Here I've inserted the rules I use locally. It includes instructions, but also one whole discussion (problem statement, details description, and then summary of what needs to be done). Hard to say to what extend it's still needed. But won't hurt.